### DISCUSSION TITLE
Bad practice to only take the files you need?

### DISCUSSION THREAD

**Discussion by cursedId0l** [Cyfrin/foundry-full-course-cu#2094] [General]
Im just wondering its bad practice to just copy the files you need into the lib folder rather than forge installing entire repos?
Like I only need the IERC20 file from open zeppelin, so can I just drag that into the lib folder rather than installing the entire open zeppelin repo?

**Comment by EngrPips** [comment]
Honestly, I am not sure if that's a bad practice or not as I just always forge install whatever I need lol but if it works for you then I think you can do it. But honestly most likely when you are building a real-world project you would have some file that imports from different files and those files also import from different it might be hard to keep copying those files into the lib folder and connecting them and all of that so then it will be more easy and logical to just install a package that contains everything you need

**Reply by cursedId0l** [reply]
I ran into the issue when I was trying to integrate uniswapv3 into a contract I'm building. Uniswap V3 was using 0.7 solidity version, and that forced me to write my contract in 0.7 since I was importing it files with 0.7. I wanted to use more modern features so I thought I would just copy the files I need and bump the compiler to 0.8. Is that common?

**Reply by EngrPips** [reply]
I haven't seen plenty of production code, so I can't really say what is common and what is not, but then if that is what you need to do to make it for now, at least you should do it, and with time, you will find out what is the most appropriate way to doing it.
